prow/flagutil/github.go

```diff
@@ -112,9 +127,16 @@ func (o *GitHubOptions) GitHubClient(secretAgent *secret.Agent, dryRun bool) (cl
 	return o.GitHubClientWithLogFields(secretAgent, dryRun, logrus.Fields{})
 }
 
+// GitHubClientWithAccessToken creates a GitHub client from an access token.
+func (o *GitHubOptions) GitHubClientWithAccessToken(token string) github.Client {
+	return github.NewClient(func() []byte { return []byte(token) }, func(content []byte) []byte {
+		return bytes.ReplaceAll(content, []byte(token), []byte("CENSORED"))
```

Comment:
This misses the case where `token` is zero-length. This was a recent bug with the other censor and not introducing it again was why I asked if we could DRY it out. If not, let's keep them in sync.

Reply:
I believe this is one you just mentioned: https://github.com/kubernetes/test-infra/pull/15076.

On my way to keep them sync.

Reply:
Done. I'll definitely take it as a TODO for myself to DRY it out if nobody is working on this.